Backpack/src/main/java/net/skyscanner/backpack/button/internal/BpkButtonBase.kt

```diff
@@ -19,30 +19,61 @@
 package net.skyscanner.backpack.button.internal
 
 import android.content.Context
+import android.content.res.ColorStateList
 import android.text.TextUtils
 import android.util.AttributeSet
 import android.view.Gravity
+import com.google.android.material.button.MaterialButton
 import net.skyscanner.backpack.R
-import net.skyscanner.backpack.text.BpkText
 
-// mainly exists here for compatibility reasons
+internal const val ICON_POSITION_START = 0
+internal const val ICON_POSITION_END = 1
+internal const val ICON_POSITION_ICON_ONLY = 2
+
 abstract class BpkButtonBase internal constructor(
   context: Context,
   attrs: AttributeSet?,
-  defStyleAttr: Int
-) : BpkButtonWithIcon(context, attrs, defStyleAttr) {
```

Comment:
BpkButtonWith icon had alsmost no userful code after migrating to MaterialButton, so I merged it.